src/main/java/uk/gov/hmcts/divorce/systemupdate/schedule/conditionalorder/SystemResendCOPronouncedCoverLettersTask.java

```diff
@@ -0,0 +1,143 @@
+                                    .must(matchQuery(String.format(DATA, APPLICANT2_OFFLINE), YES))
+                                    .must(matchQuery(String.format(DATA, APPLICANT2_PRIVATE_CONTACT), PRIVATE.getType()))
+                            )
+                            .minimumShouldMatch(1)
+                    )
+                    .mustNot(matchQuery(String.format(DATA, NOTIFICATION_FLAG), YES));
+
+            final List<CaseDetails> casesToBeUpdated =
+                ccdSearchService.searchForAllCasesWithQuery(query, user, serviceAuth, ConditionalOrderPronounced);
+
+            for (final CaseDetails caseDetails : casesToBeUpdated) {
+                triggerResendCoPronouncedCoverLetterForEligibleCases(user, serviceAuth, caseDetails);
+            }
+
+            log.info("SystemResendCOPronouncedCoverLettersTask completed.");
+        } catch (final CcdSearchCaseException e) {
+            log.error("SystemResendCOPronouncedCoverLettersTask stopped after search error", e);
+        } catch (final CcdConflictException e) {
+            log.info("SystemResendCOPronouncedCoverLettersTask stopping "
+                + "due to conflict with another running task"
+            );
+        }
+    }
+
+    private void triggerResendCoPronouncedCoverLetterForEligibleCases(User user, String serviceAuth, CaseDetails caseDetails) {
+        try {
+
+            final CaseData caseData = objectMapper.convertValue(caseDetails.getData(), CaseData.class);
+
+            if (isCaseEligibleToResendCoverLetter(caseData)) {
+                log.info("Submitting Resend CO Pronounced letter for Case {}", caseDetails.getId());
+                ccdUpdateService.submitEvent(caseDetails, SYSTEM_RESEND_CO_PRONOUNCED_COVER_LETTER, user, serviceAuth);
+            }
+        } catch (final CcdManagementException e) {
+            log.error("Submit event failed for case id: {}, continuing to next case", caseDetails.getId());
+        } catch (final IllegalArgumentException e) {
+            log.error("Deserialization failed for case id: {}, continuing to next case", caseDetails.getId());
+        }
+    }
+
+    private boolean isCaseEligibleToResendCoverLetter(final CaseData caseData) {
+        CaseDocuments caseDocuments = caseData.getDocuments();
+
+        if (caseData.getApplicant1().isOffline()
+            && caseData.getApplicant1().isConfidentialContactDetails()
+            && coverLettersAreNotUnderConfidentialList(caseDocuments, CONDITIONAL_ORDER_GRANTED_COVERSHEET_APP_1)) {
+            return true;
+        }
+
+        return caseData.getApplicant2().isOffline()
+            && caseData.getApplicant2().isConfidentialContactDetails()
+            && coverLettersAreNotUnderConfidentialList(caseDocuments, CONDITIONAL_ORDER_GRANTED_COVERSHEET_APP_2);
+    }
+
+    private boolean coverLettersAreNotUnderConfidentialList(final CaseDocuments caseDocuments, final DocumentType coverLetterType) {
+        return caseDocuments
```

Comment:
This could be added as a method of CaseDocuments

Reply:
refactored and moved to CaseDocuments